SharedPackages/BrowserServicesKit/Sources/Subscription/Managers/SubscriptionManagerV2.swift

```diff
@@ -417,10 +418,44 @@ public final class DefaultSubscriptionManagerV2: SubscriptionManagerV2 {
 
     var cachedUserEntitlements: [SubscriptionEntitlement] {
         get {
-            userDefaults.userEntitlements ?? []
+            userDefaults.userEntitlements
         }
         set {
+            let currentCachedUserEntitlements = userDefaults.userEntitlements
             userDefaults.userEntitlements = newValue
+
+            // Send notification when entitlements change
+            if !SubscriptionEntitlement.areEntitlementsEqual(currentCachedUserEntitlements, newValue) {
+                Logger.subscription.debug("Entitlements changed - New \(String(describing: newValue)) Old \(String(describing: currentCachedUserEntitlements))")
+                let payload = EntitlementsDidChangePayload(entitlements: newValue)
+                NotificationCenter.default.post(name: .entitlementsDidChange, object: self, userInfo: payload.notificationUserInfo)
+            }
+        }
+    }
+
+    var cachedIsUserAuthenticated: Bool {
```

Comment:
Ok I don't want to overcomplicate things, but could we tag this `@MainActor` and consider to push it - unless it disrupts a lot of existing code, then we can postpone.

Reply:
why @MainActor? user defaults is thread safe
I tried, but that's used in getToken and many other functions, we are opening a can of worms trying to force stuff on main thread. If needed, we should create an internal, non-main-thread, concurrency approach for achieving what we need, but that's a longer task that I would avoid today